x-pack/legacy/plugins/maps/public/layers/styles/vector_style.js

```diff
@@ -394,15 +394,18 @@ export class VectorStyle extends AbstractStyle {
 
       for (let j = 0; j < styleFields.length; j++) {
         const { supportsFeatureState, isScaled, name, range, computedName } = styleFields[j];
-        const value = parseFloat(feature.properties[name]);
+        //Date fields pulled from doc_values is an array of epoch_millis and a date string
+        const value = Array.isArray(feature.properties[name])
```

Comment:
Can you move the logic for extracting date fields out of VectorStyle which is generic and should not have source specific details. You can put the logic into [hitsToGeoJson](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/elasticsearch_geo_utils.js#L58)